From 36805044f63c2b3f396dc9af28cc401365d037e8 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 12 Nov 2012 11:30:41 +0100
Subject: [PATCH] gpio: xilinx_gpiops: Fix section mismatch

Warning log:
WARNING: vmlinux.o(.data+0x91f0): Section mismatch in reference from
the variable xgpiops_driver to the function .init.text:xgpiops_probe()
The variable xgpiops_driver references
the function __init xgpiops_probe()
If the reference is valid then annotate the
variable with __init* or __refdata (see linux/init.h) or name the variable:
*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpio/gpio-xilinxps.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpio/gpio-xilinxps.c b/drivers/gpio/gpio-xilinxps.c
index e146a65..b872977 100644
--- a/drivers/gpio/gpio-xilinxps.c
+++ b/drivers/gpio/gpio-xilinxps.c
@@ -518,7 +518,7 @@ static const struct dev_pm_ops xgpiops_dev_pm_ops = {
  * Note: Interrupts are disabled for all the banks during initialization.
  * Returns 0 on success, negative error otherwise.
  */
-static int __init xgpiops_probe(struct platform_device *pdev)
+static int __devinit xgpiops_probe(struct platform_device *pdev)
 {
 	int ret;
 	unsigned int irq_num;
-- 
1.7.0.4

